packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet/TransferTab.tsx

```diff
@@ -76,6 +96,14 @@ export function TransferTab(props: TransferTabProps) {
     }, [transferCallback])
     //#endregion
 
+    const ETH_ADDRESS = useConstant(CONSTANTS, 'ETH_ADDRESS')
+
+    const onClick = useCallback(async () => {
+        const gas = await getGas(chainId, account, ETH_ADDRESS, tokenBalance)
+        const balance = new BigNumber(formatBalance(new BigNumber(tokenBalance), token.decimals)).minus(gas).toFixed()
```

Comment:
It would help if you rolled math operations on the WEI directly. Here are many unnecessary conversions.

Reply:
```suggestion
        const balance = new BigNumber(formatBalance(tokenBalance, token.decimals)).minus(gas).toFixed()
```